cartography/intel/github/repos.py

```diff
@@ -77,19 +87,24 @@ def transform(repos_json):
     Parses the JSON returned from GitHub API to create data for graph ingestion
     :param repos_json: the list of individual repository nodes from GitHub. See tests.data.github.repos.GET_REPOS for
     data shape.
-    :return: Dict containing the repos, repo->language mapping, and owners->repo mapping
+    :return: Dict containing the repos, repo->language mapping, owners->repo mapping, and outside collaborators->repo
+    mapping.
     """
     transformed_repo_list = []
     transformed_repo_languages = []
     transformed_repo_owners = []
+    # See https://docs.github.com/en/graphql/reference/enums#repositorypermission
+    transformed_collaborators = {'ADMIN': [], 'MAINTAIN': [], 'READ': [], 'TRIAGE': [], 'WRITE': []}
     for repo_object in repos_json:
         _transform_repo_languages(repo_object['url'], repo_object, transformed_repo_languages)
         _transform_repo_objects(repo_object, transformed_repo_list)
         _transform_repo_owners(repo_object['owner']['url'], repo_object, transformed_repo_owners)
+        _transform_collaborators(repo_object['collaborators'], repo_object['url'], transformed_collaborators)
```

Comment:
I get that this is following the conventions above..

This is unnecessary mutation IMO.  You can just have `_transform_*` take in 1 less parameter and return the dict or the list.  Mutation just makes code harder to follow and it's easier to write insidious bugs once you expand on this.

Reply:
Good feedback, thanks, I'll fix the out-param behavior in a follow-up.